README.md

```diff
@@ -10,4 +10,4 @@
     -   id: check-flux-helm-values
 ```
 
-The hook depends on the helm binary being available in the path (but it doesn't require to be able to connect to a cluster).
```

Comment:
can we make kustomize optional

Reply:
No, it is required to build the fluxtomizations. If this is a non-starter and you don't want to take on this dependency, I understand.

Reply:
I think it's fine, if it's needed. Would it make sense to download it automatically if it's missing?

Reply:
I agree it makes sense to make it as easy as possible to use, including all needed dependencies.

I'm familiar with how Docker or apt might manage dependencies, but not super experienced with pre-commit hooks and how environments are typically managed. I see some languages support `additional_dependencies`, but not sure that is appropriate here (not python). I'm imagining that users may want a specific version of helm or kustomize, so the version may need to be an argument or override from a default, or maybe just a url.

Are you aware of prior art or best practice here? I went and looked at a few other k8s related hooks and they are all just native python validation, e.g. https://github.com/willthames/kubernetes-validate (side node, i'm tempted to add more options to run those validations for future work)

Reply:
no, unfortunately I don't know either. What I was wondering if we can do something simple like checking if the binary is in the path and if not download it. But there might be a better way.

Reply:
To me, it feels a bit invasive to install the tools on behalf of the user.

Isn't it likely they'll have many of them installed? It's probably not too difficult to install any missing tools, given the target audience [1]. On MacOS and Linux, it could be a one-liner:

```
brew install kustomize helm fluxcd/tap/flux kyverno
```

I do like the option of using Docker for an isolated, pre-built environment. This could also be useful as part of a Github Action. That being said, I know [megalinter](https://megalinter.io/latest/mega-linter-runner/) uses Docker locally, and I found it to be a bit painful to use. It's also slower. Perhaps Docker could be an alternative mode?

As a short-term solution, what do you think about a startup check to ensure the tools are installed? If any of the tools are missing, it could fail with a link to documentation?

---
[1] Feel free to disagree if this is an unconscious bias of mine. 😄

Reply:
You're right, that's probably good enough

Reply:
you could provide 3 hooks (simple as it is, with dependencies installed local, and docker)